modules/yandexIdSystem.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * The {@link module:modules/userId} module is required
+ * @module modules/yandexIdSystem
+ * @requires module:modules/userId
+ */
+
+import { MODULE_TYPE_UID } from '../src/activities/modules.js';
+import { submodule } from '../src/hook.js';
+import { getStorageManager } from '../src/storageManager.js';
+import { logInfo } from '../src/utils.js';
+
+const BIDDER_CODE = 'yandex';
+const BIDDER_EID_KEY = 'yandex.com';
+const YANDEX_ID_KEY = 'yandexId';
+
+const USER_ID_KEY = '_ym_uid';
+const USER_ID_COOKIE_EXP_MS = 31536000000; // 365 days
+
+export const storage = getStorageManager({
+  moduleType: MODULE_TYPE_UID,
+  moduleName: BIDDER_CODE,
+});
+
+export const yandexIdSubmodule = {
+  /**
+   * Used to link submodule with config.
+   * @type {string}
+   */
+  name: BIDDER_CODE,
+  /**
+   * Decodes the stored id value for passing to bid requests.
+   * @param {string} value
+   */
+  decode(value) {
+    logInfo('decoded value yandexId', value);
+
+    return { [YANDEX_ID_KEY]: value };
+  },
+  getId() {
+    const yandexUidStorage = new YandexUidStorage(storage);
+
+    if (!yandexUidStorage.checkIsAvailable()) {
+      return;
+    }
+
+    return {
+      id: yandexUidStorage.getUid(),
+    };
+  },
+  eids: {
+    [YANDEX_ID_KEY]: {
+      source: BIDDER_EID_KEY,
+      atype: 1,
+    },
+  },
+};
+
+class YandexUidStorage {
+  /**
+   * @param {typeof cookieStorage} cookieStorage
+   */
+  constructor(cookieStorage) {
+    this._cookieStorage = cookieStorage;
+  }
+
+  _generateUid() {
+    return new YandexUidGenerator().generateUid();
+  }
+
+  _getUserIdFromStorage() {
+    const id = this._cookieStorage.getCookie(USER_ID_KEY);
+
+    return id;
+  }
+
+  _setUid(userId) {
+    if (this._cookieStorage.cookiesAreEnabled()) {
+      const expires = new Date(Date.now() + USER_ID_COOKIE_EXP_MS).toString();
+
+      this._cookieStorage.setCookie(USER_ID_KEY, userId, expires);
```

Comment:
The parent module can handle setting the id accorring to publisher config, the submodules only needs to gather it

Reply:
See #10710

Reply:
Hi @patmmccann, thanks for the recommendation! **Is it required by Prebid for us to maintain different types of storage and allow custom keys?** Can we just fall back to "do nothing"+"log error" when `storage` option is customized for our submodule?

I'm asking this because this submodule is intended to be a thin bridge between our Yandex Metrika publisher tag and our bidder adapter. And we want it to be kept as simple as possible.

We intentionally want only to use the cookie and constant `USER_ID_KEY`.